haystack/nodes/retriever/link_content.py

```diff
@@ -43,31 +55,67 @@ class LinkContentFetcher(BaseComponent):
 
     LinkContentFetcher supports the following content types:
     - HTML
+    - PDF
 
     """
 
     outgoing_edges = 1
 
+    USER_AGENT = f"haystack/LinkContentRetriever/{__version__}"
+
     REQUEST_HEADERS = {
         "accept": "*/*",
-        "User-Agent": f"haystack/LinkContentFetcher/{__version__}",
+        "User-Agent": USER_AGENT,
         "Accept-Language": "en-US,en;q=0.9,it;q=0.8,es;q=0.7",
         "referer": "https://www.google.com/",
     }
 
-    def __init__(self, processor: Optional[PreProcessor] = None, raise_on_failure: Optional[bool] = False):
+    def __init__(
+        self,
+        processor: Optional[PreProcessor] = None,
+        raise_on_failure: Optional[bool] = False,
+        user_agents: Optional[List[str]] = None,
+        retry_attempts: Optional[int] = None,
+    ):
         """
 
         Creates a LinkContentFetcher instance.
         :param processor: PreProcessor to apply to the extracted text
         :param raise_on_failure: A boolean indicating whether to raise an exception when a failure occurs
                          during content extraction. If False, the error is simply logged and the program continues.
                          Defaults to False.
+        :param user_agents: A list of user agents to use when fetching content. Defaults to None.
+        :param retry_attempts: The number of times to retry fetching content. Defaults to 2.
         """
         super().__init__()
         self.processor = processor
         self.raise_on_failure = raise_on_failure
-        self.handlers: Dict[str, Callable] = {"html": html_content_handler, "pdf": pdf_content_handler}
+        self.user_agents = itertools.cycle(user_agents or [LinkContentFetcher.USER_AGENT])
+        self.default_user_agent = next(self.user_agents)
```

Comment:
I think it makes sense to default back to the default agent in the current design. During request handling issues (mainly blocking), we can cycle through alternate user agents and see if we can get the content but then always default back to the default. I should document this in fact.

Reply:
Why not simply going back to the start of the list though? That's what the code does anyway: it saves the first item of the list in this attribute and goes back to it every time, if I understood it right. Therefore we could simply use the original list instead of duplicating the value.